File: browser-integrations/kernel/typescript/src/status.ts
/**
 * Lightweight progress output for CLI runs.
 *
 * Writes timestamped phase lines to stderr so they show live while a command
 * runs, keeping stdout reserved for the final structured result.
 */

const START = Date.now();

/** Print one timestamped progress line to stderr. */
export function status(message: string): void {
  const elapsed = ((Date.now() - START) / 1000).toFixed(1).padStart(5);
  process.stderr.write(`  [${elapsed}s] ${message}\n`);
}

File: browser-integrations/kernel/python/kernel_runloop/status.py
"""Lightweight progress output for CLI runs.

Writes timestamped phase lines to stderr so they show live while a command runs,
keeping stdout reserved for the final structured result.
"""

import sys
import time

_START = time.monotonic()


def status(message: str) -> None:
    """Print one timestamped progress line to stderr."""
    elapsed = time.monotonic() - _START
    print(f"  [{elapsed:5.1f}s] {message}", file=sys.stderr, flush=True)
